pkg/sources/v1alpha2/ping_client.go

```diff
@@ -87,7 +89,15 @@ func (c *pingSourcesClient) DeletePingSource(name string) error {
 }
 
 func (c *pingSourcesClient) GetPingSource(name string) (*v1alpha2.PingSource, error) {
-	return c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	source, err := c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, knerrors.GetError(err)
+	}
+	err = updateSourceGVK(source)
+	if err != nil {
+		return nil, knerrors.GetError(err)
+	}
```

Comment:
```suggestion
	err = updateSourceGVK(source)
	if err != nil {
		return nil, err
	}
```

Reply:
this is only client-side operation and we can simply return the `err` here

Although `knerrors.GetError` would just get back with original error, the idea behind passing `err` through `knerrors.GetError` is to catch errors from server for eg: if the underlying resource isn't present / serving or eventing isn't installed.